src/types.js

```diff
@@ -54,7 +54,7 @@ export type ActorId = string;
  * @static
  */
 export type Location = {
-  sourceId: SourceId,
+  sourceId?: SourceId,
```

Comment:
sorry, this shouldnt be optional, the param for the reducer should be

Reply:
This is still a problem if I change this back and edit the reducer param. 
In ast.js, location is declared multiple times and if sourceId is a required param then it should be included.. seperated... or made optional

Reply:
![image](https://user-images.githubusercontent.com/23143862/38718353-edfd2630-3ea9-11e8-9789-e61d300b2bcd.png)